### PR TITLE
[web] Fix inability to type in text fields in iOS

### PR DESCRIPTION
**Pull request by mdebbar**
In iOS Safari, we should do the same focus re-acquisition logic that we do for Android.

The only downside that this PR introduces is that the flutter text field remains focused after the user clicks "done" on the keyboard (filed an [issue](https://github.com/flutter/flutter/issues/86682)). This isn't a big deal because the user can just tap the text field again to show the keyboard.

Fixes https://github.com/flutter/flutter/issues/84817

### CODE REVIEW

File: lib/web_ui/lib/src/engine/text_editing/text_editing.dart [hunk 1]
```diff
@@ -1133,13 +1133,23 @@ class IOSTextEditingStrategy extends GloballyPositionedTextEditingStrategy {
...
-    // [domRenderer.windowHasFocus] is not checked in [IOSTextEditingStrategy].
+    // 1. The browser app is sent to the background (or the tab is changed). In
+    //    this case, the window loses focus (see [domRenderer.windowHasFocus]),
+    //    so we close the input connection with the framework.
+    // 2. The user taps on another focusable element. In this case, we refocus
+    //    the input field and wait for the framework to manage the focus change.
+    // 3. The virtual keyboard is closed by tapping "done". We can't detect this
+    //    programmatically, so we end up refocusing the input field. This is
+    //    okay because the virtual keyboard will hide, and as soon as the user
+    //    taps the text field again, the virtual keyboard will up.
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/text_editing/text_editing.dart:1146]: nit: "will _come_ up"?

File: lib/web_ui/test/text_editing_test.dart [hunk 2]
```diff
@@ -584,24 +584,12 @@ void testMain() {
...
-        // DOM element loses the focus.
-        expect(defaultTextEditingRoot.activeElement, null);
-      } else {
-        // No connection close message sent.
-        expect(spy.messages, hasLength(0));
-        await Future<void>.delayed(Duration.zero);
-        // DOM element still keeps the focus.
-        expect(defaultTextEditingRoot.activeElement,
-            textEditing!.strategy.domElement);
-      }
```
Comment (yjbanov) [inline lib/web_ui/test/text_editing_test.dart:604]: Nice! Another browser engine if/else gone! 👍

**Review by yjbanov — APPROVED** [review]
lgtm

File: lib/web_ui/lib/src/engine/text_editing/text_editing.dart (hunk 1 above)
Comment (mdebbar) [reply lib/web_ui/lib/src/engine/text_editing/text_editing.dart:1146]: Nice catch! I meant to say "show up" but "come up" is also good :)

File: lib/web_ui/test/text_editing_test.dart (hunk 2 above)
Comment (richard457) [reply lib/web_ui/test/text_editing_test.dart:604]: Awesome!

**Pull request merged**